src/parser/src/main/java/io/github/syst3ms/skriptparser/sections/SecFilter.java

```diff
@@ -0,0 +1,62 @@
+package io.github.syst3ms.skriptparser.sections;
+
+import io.github.syst3ms.skriptparser.Main;
+import io.github.syst3ms.skriptparser.lang.Expression;
+import io.github.syst3ms.skriptparser.lang.Statement;
+import io.github.syst3ms.skriptparser.lang.TriggerContext;
+import io.github.syst3ms.skriptparser.lang.Variable;
+import io.github.syst3ms.skriptparser.lang.lambda.ReturnSection;
+import io.github.syst3ms.skriptparser.lang.lambda.SkriptFunction;
+import io.github.syst3ms.skriptparser.parsing.ParseContext;
+import io.github.syst3ms.skriptparser.types.changers.ChangeMode;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public class SecFilter extends ReturnSection<Boolean> {
+    private Variable<?> filtered;
+    private SkriptFunction<SecFilter, Boolean> lambda;
+
+    static {
+        Main.getMainRegistration().addSection(
+                SecFilter.class,
+                "filter %^objects%"
+        );
+    }
```

Comment:
Why only variables? I can consider some cases where I'd want to use normal objects.
Edit: Adding this would need checks for the ChangeMode and other things. While this would need several changes just to support one feature, I think it's worth it.

Reply:
Right, I forgot things other than variables can be set.